lib/tree.rb

```diff
@@ -16,47 +16,121 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log(n)) where n is the number of nodes
+  # Space Complexity: O(1)
```

Comment:
Remember since you're doing this recursively, the system stack will allocate space for each recursive call.  

So if the tree is balanced the space complexity is O(log n), if it's widely unbalanced O(n).  The same also applies to time complexity.